core/lib/src/form/validate.rs

```diff
@@ -228,6 +231,9 @@ pub fn neq<'v, A, B>(a: &A, b: B) -> Result<'v, ()>
 ///
 /// [`form::Result<'_, T>`]: crate::form::Result
 pub trait Len<L> {
+    /// Check if the value is empty in constant time.
+    fn is_empty(&self) -> bool;
```

Comment:
Why are we adding this?

Reply:
It is to fix the warning [len_without_is_empty](https://rust-lang.github.io/rust-clippy/master/index.html#len_without_is_empty). They say that, for some containers, `is_empty` is much faster because it is done in constant time instead of `len` which will be in linear time.

To be honest, I don't know if it is useful in this context because unlike C++'s STL, Rust API don't tell the complexity of accessing a container. Moreover I did not check if I could replace some `len` with `is_empty`.

For now, all the lints in this commit will be allowed and I'll fix them in another PR. I will check whether this function will be used or not. If it is useful, I'll replace `len() == 0` with `is_empty()`, otherwise I will allow this lint. If you prefer not implementing this function at all, please tell it to me : I don't want to waste my time on things that won't be merged.